docs/build/smart-contracts/reference/library.md

```diff
@@ -0,0 +1,668 @@
+
+### path_extract/2
+
+```elixir
+Json.path_extract("{ \"firstName\": \"John\", \"lastName\": \"Doe\"}", "$.firstName") # "John"
+```
+
+Parameters:
+- `json` the json string
+- `path` the path
+
+Returns the data at `path` in `json`.
+
+### path_match?/2
+
+```elixir
+Json.path_match?("{\"1622541930\":{\"uco\":{\"eur\":0.176922,\"usd\":0.21642}}}", "$.*.uco.usd") # true
+Json.path_match?("{\"1622541930\":{\"uco\":{\"eur\":0.176922,\"usd\":0.21642}}}", "$.*.uco.aud") # false
+```
+
+Parameters:
+- `json` the json string
+- `path` the path
+
+Returns if there is a value at `path` in `json`.
+
+### to_string/1
+
+```elixir
+Json.to_string(1)           # "1"
+Json.to_string("str")       # "\"str\""
+Json.to_string([1,2,3])     # "[1,2,3]"
+Json.to_string(foo: "bar")  # "{\"foo\":\"bar\"}"
+```
+Parameters:
+- `any` anything
+
+Returns the Json representation of `any`.
+
+### is_valid?/1
+
+```elixir
+Json.is_valid?("str")        # false
+Json.is_valid?("\"str\"")    # true
+```
+Parameters:
+- `json` the json string
+
+Returns whether the string is valid JSON or not.
+
+---------
+
+## Time
+
+### now/0
+
+```elixir
+Time.now()  # 1677493444
+```
+
+Returns the current unix time (seconds since epoch).
+
+:::danger 
+[THIS REQUIRE SOME CHANGE TO BE DETERMINIST]
+:::
+
+---------
+
+## Chain 
+
+### get_genesis_address/1
+
+```elixir
+Chain.get_genesis_address("000012345...") # "000056789..."
+Chain.get_genesis_address("000012345...") # "000012345..."
+```
+
+Parameters:
+- `address` the transaction address
+
+Returns the genesis address of the transaction chain that contains a transaction at `address`. If there is no transaction at `address`, it returns `address`.
+
+### get_first_transaction_address/1
+
+```elixir
+Chain.get_first_transaction_address("000012345...") # "000056789..."
+```
+
+Parameters:
+- `address` the transaction address
+
+Returns the address of the first transaction of the transaction chain that contains a transaction at `address`.
+
+:::danger
+[IT CURRENTLY CRASH IF THERE IS NO TRANSACTION]
+:::
+
+### get_genesis_public_key/1
+
+```elixir
+Chain.get_genesis_public_key("00013B08D...") # "000108A5C..."
+```
+
+Parameters:
+- `public_key` the public key
+
+Returns the genesis public key of `public_key`.
+
+---------
+
+## Crypto
+
+### hash/1
+
+Equivalent to [hash/2](#hash2) with the second argument "sha256".
+
+### hash/2
+
+```elixir
+Crypto.hash("my text", "sha256") # "7330D2B39CA35EA..."
+Crypto.hash("my text", "sha512") # "A3D7E45A0D9BE5F..."
+```
+
+Parameters:
+- `str` the text to hash
+- `algo` the hash algorithm ("sha256", "sha512", "sha3_256", "sha3_512", "blake2b")
+
+Returns the hash of `str` by using `algo` algorithm.
+
+---------
+
+## Token
+
+### hash/2
+
+```elixir
+Token.fetch_id_from_address("000012345") # "3C21EF708BF11B4232..."
+```
+
+Parameters:
+- `address` the token address
+
+Returns the token's id of the token at `address`.
+
+---------
+
+## Contract
```

Comment:
Maybe contrat should be explained also in actions part to explain how to build the next transaction